src/metrics/metrics.ts

```diff
@@ -492,6 +484,44 @@ export async function getPieceLevelMoveInfo(games: FileReaderGame[]) {
     piecesWithMostMovesInAGame,
     gameLinksWithPiecesMostMoves,
     numMovesMadePieceWithMostMoves,
+  };
+}
+
+export async function getPiecePromotionInfo(games: FileReaderGame[]) {
+  let ambigPiecePromotedToMap = {};
+  let promotingPieceMap = {};
+
+  for (const game of games) {
+    const chess = new Chess();
+    chess.loadPgn(game.moves);
+    const chessHistory = chess.history();
+
+    for (const moveInfo of chessHistory) {
+      if (moveInfo.originalString.includes('=')) {
+        // REGEX to only capture the piece type
+        const piecePromotedTo = moveInfo.originalString
+          .split('=')[1]
+          .match(/[a-zA-Z]/)[0];
+
+        const promotingPiece = moveInfo.unambiguousSymbol;
```

Comment:
why are we doing this? Is this information not included in the history? Aren't there flags on prettyMove that specify if a promotion occurs, and to what?

Reply:
Yes - good point, flags are included in prettyMove. can refactor later to include prettyMove in these. but not 100% necessary for V1 analysis

Reply:
let's just fix this now. It will be easier to understand and we won't be recompiling a regex

Reply:
and it should be simple